### ISSUE TITLE
Large explosions on shuttle in transit can still nuke the server

### ISSUE THREAD

**Issue by ValkyrieSkies** [d3athrow/vgstation13#35085]
#### Description of issue
A poopmin set off a 2000x2000 (20000 dev) explosion on the escape shuttle in transit and broke the server, requiring a brute force restart go get things running again. Admins might recall there was a similar issue in the past where someone firing a blast cannon on the shuttle in transit also crashed the server. This was previously thought to have been fixed but it's gone and reared its ugly head again.

I initially remarked that we should probably put a cap on admin custom explosions, but judging by Exxion's remarks the ability to make a 2000x2000 explosion isn't actually that dire so  it can be ignored.

#### Server revision
f40bd5f75074f0e6b9d2631a138a9d7ba6fc1884 

[administration] [general]

**Comment by ValkyrieSkies** [comment]
Also pinged Deity specifically because he dealt with the prior issue as well.

**Comment by Eneocho** [comment]
`>putting guardrails on admeme tools`
If the shitmin wants to go out with a bang and blow a 2000dev bomb, let him.
Let god sort em out.

**Comment by ValkyrieSkies** [comment]
> `>putting guardrails on admeme tools` If the shitmin wants to go out with a bang and blow a 2000dev bomb, let him. Let god sort em out.

Given that it causes a server crash that has in the past been very difficult to reboot from, it is something that should be addressed.

**Comment by boy2mantwicethefam** [comment]
Could add an emergency button for stopping all ongoing explosions.

**Comment by Hinaichigo** [comment]
If you do cap it maybe make it like at least several times higher than the nuke like 10 times or so or whatever's reasonable.

**Comment by Exxion** [comment]
There's no obvious reason any explosion larger than a Z-level should take longer than any other
So like 500 dev and 1000000 dev should be basically equivalent, since each of them only has 250000 total tiles to do anything to
If there *is* a difference, that should be fixable

EDIT: It also sounds like there's an issue specifically with explosions in shuttle zones, maybe?

**Comment by ValkyrieSkies** [comment]
> There's no obvious reason any explosion larger than a Z-level should take longer than any other So like 500 dev and 1000000 dev should be basically equivalent, since each of them only has 250000 total tiles to do anything to If there _is_ a difference, that should be fixable
> 
> EDIT: It also sounds like there's an issue specifically with explosions in shuttle zones, maybe?

Yeah, there's some sort of fringe problem involving how explosions react with the warp tiles in shuttle zones. Deity helped address it last time it came up with the blast cannon scenario, which is why I'm invoking him again.

**Comment by ValkyrieSkies** [comment]
Heavily reworded to more properly reflect what the actual issue is.
